internal/controller/providers/psmdb/provider.go

```diff
@@ -147,6 +154,32 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 	return status, nil
 }
 
+func isPVCResizeInProgress(ctx context.Context, c client.Client, psmdb *psmdbv1.PerconaServerMongoDB) (bool, error) {
+	if psmdb.Status.State == psmdbv1.AppStateInit {
+		stsList := &appsv1.StatefulSetList{}
+		err := c.List(
+			ctx,
+			stsList,
+			client.InNamespace(psmdb.GetNamespace()),
+			client.MatchingLabels{
+				"app.kubernetes.io/instance": psmdb.GetName(),
+			},
+		)
+		if err != nil {
+			return false, err
+		}
+		for _, sts := range stsList.Items {
+			annots := sts.GetAnnotations()
+			_, ok := annots[psmdbv1.AnnotationPVCResizeInProgress]
+			if ok {
+				return true, nil
+			}
+		}
+		return false, nil
```

Comment:
just a nitpick - looks like this return can be omitted

Reply:
Leftovers from some refactoring, updated!